src/sql/src/plan/statement/ddl.rs

```diff
@@ -2724,46 +2725,32 @@ const DEFAULT_INTROSPECTION_GRANULARITY: Interval = Interval {
     days: 0,
 };
 
+generate_extracted_config!(
+    ReplicaOption,
+    (AvailabilityZone, String),
+    (Size, String),
+    (Remote, Vec<String>)
```

Comment:
Will it be possible/easy to extend this macro to support required parameters? I think for `ReplicaOption` everything is optional, but that won't always be the case!

Reply:
A few thoughts here: this is built to extract the values from the options--seems like additional checks  can be handled somewhat trivially on the return value, so don't know if it's worth contorting this macro to do that.

That being said, we could e.g. add an optional third field like `Required`, and inline some kind of check that we've seen the named key. The challenge there is just that the declarative macro could get messy if we want to start adding additional/arbitrary constraints.

Reply:
What is/was awesome about the existing macro is that required fields are modeled as `T` while optional fields are modeled as `Option<T>` in the generated struct. That means that the required-ness just falls out of the type system; you can't forget to come up with a default for an optional field.

Reply:
Ah, yeah, ultimately we'll want to support both required options and default values. I'm working on an extension to the macro to offer both of those features.